optuna/terminator/callback.py

```diff
@@ -13,6 +13,55 @@
 
 @experimental_class("3.2.0")
 class TerminatorCallback:
+    """A callback that terminates the optimization using Terminator.
+
+    This class implements a callback which wraps :class:`~optuna.terminator.Terminator`
+    so that it can be used with the :func:`~optuna.study.Study.optimize` method.
+
+    Args:
+        terminator:
+            A terminator object which determines whether to terminate the optimization by
+            assessing the room for optimization and statistical error. Defaults to a
+            :class:`~optuna.terminator.Terminator` object with default
+            improvement_evaluator and error_evaluator.
+
+    Example:
```

Comment:
Maybe we can add another example to show how `optuna.terminator` module can be used when no cross-validation is performed?

Reply:
I think it can be done as a follow-up.